CONTRIBUTING.md

```diff
@@ -1,69 +1,216 @@
+file where it is used.
+
+Our naming structure uses kebab case; for example, `my-file-name.rst`.
+
+## Supported formats and syntax
+
+Our documentation includes both Markdown and ReST files. We are gradually transitioning existing
+Markdown to ReST in order to more effectively meet our documentation needs. When contributing,
+ReST is preferred; if you must use Markdown, use GitHub-flavored Markdown.
+
+We use [Sphinx Design](https://sphinx-design.readthedocs.io/en/latest/index.html) syntax and compile
+our API references using Doxygen.
+
+The following table shows some common documentation components and the syntax convention we
+use for each:
+
+<table>
+<tr>
+<th>Component</th>
+<th>ReST syntax</th>
+</tr>
+<tr>
+<td>Code blocks</td>
+<td>
+
+```rst
+
+..  code-block:: language-name
+
+My code block.
+
+
+```
+
+</td>
+</tr>
+<tr>
+<td>Cross-referencing internal files</td>
+<td>
+
+```rst
+
+:doc:`Title <../path/to/file/filename>`
+
+```
+
+</td>
+</tr>
+<tr>
+<td>External links</td>
+<td>
+
+```rst
+
+`link name  <URL>`_
+
+```
+
+</td>
+</tr>
+<tr>
+<tr>
+<td>Headings</td>
+<td>
+
+```rst
+
+******************
+Chapter title
+******************
+
+Section title
+===============
+
+Subsection title
+---------------------
+
+Sub-subsection title
+^^^^^^^^^^^^^^^^^^^^
+
+
+```
+
+</td>
+</tr>
+<tr>
+<td>Internal links</td>
+<td>
 
-Adopt Microsoft CPP-Docs guidelines for
-[Voice and Tone](https://github.com/MicrosoftDocs/cpp-docs/blob/main/styleguide/voice-tone.md).
+```rst
+
+1. Add a tag to the section you want to reference:
+
+.. _my-section-tag: section-1
+
+Section 1
+==========
+
+2. Link to your tag:
+
+As shown in :ref:`section-1`.
+
+```
+
+</td>
+</tr>
+<tr>
+<tr>
+<td>Lists</td>
+<td>
+
+```rst
+
+# Ordered list item
+
+* Unordered list item
+
+```
+
+</td>
+</tr>
+<tr>
+<tr>
+<td>Math (block)</td>
+<td>
+
+```rst
+
+.. math::
+A = \begin{pmatrix}
+        0.0 & 1.0 & 1.0 & 3.0 \\
+        4.0 & 5.0 & 6.0 & 7.0 \\
+      \end{pmatrix}
+
+```
+
+</td>
+</tr>
+<tr>
+<td>Math (inline)</td>
+<td>
+
+```rst
+
+:math:`2 \times 2 `
+
+```
+
+</td>
+</tr>
+<tr>
+<td>Notes</td>
+<td>
+
+```rst
+
+.. note::
+
+My note here.
+
+```
+
+</td>
+</tr>
+<tr>
+<td>Tables</td>
+<td>
+
+```rst
+
+.. csv-table::  Optional title here
+    :widths: 30, 70  #optional column widths
+   :header: "entry1 header", "entry2 header"
+
+   "entry1", "entry2"
+
+```
+
+</td>
+</tr>
+</table>
+
+## Language and style
 
-ROCm documentation templates to be made public shortly. ROCm templates dictate
-the recommended structure and flow of the documentation. Guidelines on how to
-integrate figures, equations, and tables are all based off
-[MyST](https://myst-parser.readthedocs.io/en/latest/intro.html).
+We use
+[Microsoft CPP-Docs](https://github.com/MicrosoftDocs/cpp-docs/blob/main/styleguide/voice-tone.md)
+guidelines for voice and tone, and
+[Microsoft style guidelines](https://learn.microsoft.com/en-us/style-guide/) for language and style.
```

Comment:
Some SMEs use the Google Style Guide. Google is the industry standard for open source documentation. That said, I am open to all options.

Reply:
Sure, I can change that. I think they're about the same. Do we still want to keep the CPP reference for voice/tone?

Reply:
I am good with whatever you decide. Here's the link to Google's cpp guide:

https://google.github.io/styleguide/cppguide.html

Reply:
I had a look at the Google guide and it doesn't really address text/language...pretty specific to writing c++. Google's writing resources page refers to the Microsoft style guide, so I think it makes sense to use that as our writing reference.

Reply:
Sure. The link I sent you was for cpp only. Here's a comprehensive list of all style guides:

https://google.github.io/styleguide/

Microsoft is good, too. Not sure how 'open source' friendly it is though.

Reply:
I like how crisp and concise this is,

https://developers.google.com/style/highlights

Reply:
I like it too. Do we want to commit to using that as our 'official' guide?